### PR TITLE
Strip whitespace in TOPLEVEL name to prevent bugs

### LINKED ISSUE #2448: TOPLEVEL should be whitespace stripped
From here: https://github.com/cocotb/cocotb/blob/ecb43878a6af0605fdf08c2f1829cc401bfdeb36/cocotb/__init__.py#L219-L225

On Questa with the FLI, trailing whitespace causes the toplevel entity to not be found since it does a strict `!strcmp()`.

### PR DESCRIPTION
**Pull request by ktbarrett**
Closes #2448.

### CODE REVIEW

**Review by marlonjames — APPROVED** [review]

**Review by cmarqu — APPROVED** [review]

**Comment by eric-wieser** [comment]
I think a note in the docs saying whitespace is stripped might be a good idea.

**Comment by eric-wieser** [comment]
To be clear, I mean a comment in the documentation for environment variables, not a release note. I probably wouldn't consider this a bugfix anyway - It's just making things easier for users who are sloppy with their makefiles (which isn't to say I'm opposed to it).

File: documentation/source/building.rst [hunk 1]
```diff
@@ -47,6 +47,7 @@ Cocotb
 
     Use this to indicate the instance in the hierarchy to use as the :term:`DUT`.
     If this isn't defined then the first root instance is used.
+    Leading and trailing whitespace are automaticall discarded.
```
Comment (eric-wieser) [inline documentation/source/building.rst:50]:
```suggestion
    Leading and trailing whitespace are automatically discarded.
```

Comment (eric-wieser) [reply documentation/source/building.rst:50]: Could add `..versionchanged` too

**Pull request merged**